plugins/woocommerce/src/Internal/Admin/Settings/PaymentProviders.php

```diff
@@ -152,6 +152,56 @@ function ( $gateway ) {
 		return $payment_gateways;
 	}
 
+	/**
+	 * Get the payment gateway provider instance.
+	 *
+	 * @param string $gateway_id The gateway ID.
+	 *
+	 * @return PaymentGateway The payment gateway provider instance.
+	 *                        Will return the general provider of no specific provider is found.
+	 */
+	public function get_payment_gateway_provider_instance( string $gateway_id ): PaymentGateway {
```

Comment:
Made this public so it can be used throughout.

Reply:
https://github.com/woocommerce/woocommerce/issues/57039